src/Control/Carrier/Class.hs

```diff
@@ -43,3 +57,80 @@ instance Monoid w => Carrier (Writer w) ((,) w) where
   eff (Tell w (w', k))    = (mappend w w', k)
   eff (Listen m k)        = uncurry k m
   eff (Censor f (w, a) k) = let (w', a') = k a in (mappend (f w) w', a')
+
+
+-- transformers
+
+instance (Carrier sig m, Effect sig) => Carrier (Error e :+: sig) (Except.ExceptT e m) where
+  eff (L (Throw e))     = Except.throwE e
+  eff (L (Catch m h k)) = Except.catchE m h >>= k
+  eff (R other)         = Except.ExceptT $ eff (handle (Right ()) (either (pure . Left) Except.runExceptT) other)
+
+instance Carrier sig m => Carrier (Reader r :+: sig) (Reader.ReaderT r m) where
+  eff (L (Ask       k)) = Reader.ask >>= k
+  eff (L (Local f m k)) = Reader.local f m >>= k
+  eff (R other)         = Reader.ReaderT $ \ r -> eff (hmap (flip Reader.runReaderT r) other)
+
+newtype RWSTF w s a = RWSTF { unRWSTF :: (a, s, w) }
+  deriving (Functor)
+
+toRWSTF :: Monoid w => w -> (a, s, w) -> RWSTF w s a
+toRWSTF w (a, s, w') = RWSTF (a, s, mappend w w')
+
+instance (Carrier sig m, Effect sig, Monoid w) => Carrier (Reader r :+: Writer w :+: State s :+: sig) (RWS.CPS.RWST r w s m) where
+  eff (L (Ask       k))      = RWS.CPS.ask >>= k
+  eff (L (Local f m k))      = RWS.CPS.local f m >>= k
+  eff (R (L (Tell w k)))     = RWS.CPS.tell w *> k
+  eff (R (L (Listen m k)))   = RWS.CPS.listen m >>= uncurry (flip k)
+  eff (R (L (Censor f m k))) = RWS.CPS.censor f m >>= k
+  eff (R (R (L (Get   k))))  = RWS.CPS.get >>= k
+  eff (R (R (L (Put s k))))  = RWS.CPS.put s *> k
+  eff (R (R (R other)))      = RWS.CPS.rwsT $ \ r s -> unRWSTF <$> eff (handle (RWSTF ((), s, mempty)) (\ (RWSTF (x, s, w)) -> toRWSTF w <$> RWS.CPS.runRWST x r s) other)
```

Comment:
> My name is Ozymandias, King of Kings;
> Look on my Works, ye Mighty, and despair!

I may never understand what `RWS` is for, but at least I know how to write a `Carrier` instance for it.

Reply:
Bro, how have we never talked about this? I’ve run _businesses_ atop RWS.

Reply:
Okay maybe that’s a slight exaggeration. I have used RWS in production. It’s good. And we have an issue about this (#162) that this should address.

Reply:
I would like to know more!